go/vt/vtgate/engine/fk_cascade.go

```diff
@@ -29,9 +29,17 @@ import (
 // FkChild contains the Child Primitive to be executed collecting the values from the Selection Primitive using the column indexes.
 // BVName is used to pass the value as bind variable to the Child Primitive.
 type FkChild struct {
+	// BVName is the bind variable name for the tuple bind variable used in the primitive.
 	BVName string
-	Cols   []int // indexes
-	Exec   Primitive
+	// Cols are the indexes of the column that need to be selected from the SELECT query to create the tuple bind variable.
+	Cols []int
+	// UpdateExprBvNames is the list of bind variables for non-literal expressions in UPDATES.
+	UpdateExprBvNames []string
+	// UpdateExprCols is the list of indexes for non-literal expressions in UPDATES that need to be taken from the SELECT.
+	UpdateExprCols []int
+	// CompExprCols is the list of indexes for the comparison in the SELECTS to know if the column is actually being updated or not.
+	CompExprCols []int
```

Comment:
can this become part of a struct and have a single slice if they are of same lenght?

Reply:
Done